DDC/DDC.swift

```diff
@@ -545,7 +553,21 @@ public class DDC {
                portSerialNumber, displaySerialNumber)
         continue
       }
-
+      
+      if unitDetect, let displayLocation = dict[kIODisplayLocationKey] as? NSString {
+        // the unit number is the number right after the last "@" sign in the display location
+        if let regex = try? NSRegularExpression(pattern: "@([0-9]+)[^@]+$", options: []) {
```

Comment:
Given the regex is static, this should be fine to avoid one level of nesting:

```suggestion
        let regex = NSRegularExpression(pattern: "@([0-9]+)[^@]+$", options: [])!
```

Reply:
Also, maybe include an example in the comment of what exactly this matches.